tests/python/pants_test/test_base.py

```diff
@@ -616,6 +616,23 @@ def assertInFile(self, string, file_path):
       content = f.read()
       self.assertIn(string, content, '"{}" is not in the file {}:\n{}'.format(string, f.name, content))
 
+  @contextmanager
+  def assertRaisesWithMessageContaining(self, exception_type, error_text, substring_match=False):
```

Comment:
Hm, this new parameter name isn't immediately clear to me what it means. I strongly prefer `exact=True`, followed by `inexact=False` instead. If those sound off, maybe `exact_message_match` or a variant.

I think `substring` is throwing me off because I always associate a substring with grabbing a specific index from the bigger string and then checking if that specific substring is present. 

--

Thanks for changing the default behavior!

Reply:
My concern was that `exact` may connote that the alternative is some sort of fuzzy matching. I was also thinking of `only_contains`. I'll push a commit using `exact=True` and see what that looks like.